From 5eca761c60919e8a39bf6a4463cb44959a1e7a3e Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Thu, 18 Dec 2014 20:08:56 +0530
Subject: [PATCH] ASoC: Intel: Defer sst platform driver probe

Sst platform driver loads the dfw bin from filesystem.
We need to defer sst platfrom driver probe until filesystem is up.

Issue: GMINL-3679

Change-Id: I08eac78d01186b2b89a3383c1a5bf544c0086a44
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
Signed-off-by: Praveen Diwakar <praveen.diwakar@intel.com>
Signed-off-by: Fabien Marotte <fabien.marotte@intel.com>
---
 sound/soc/intel/pcm.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/soc/intel/pcm.c b/sound/soc/intel/pcm.c
index 7931691..ac3a87b 100644
--- a/sound/soc/intel/pcm.c
+++ b/sound/soc/intel/pcm.c
@@ -944,8 +944,16 @@ static int sst_platform_probe(struct platform_device *pdev)
 	struct sst_data *sst;
 	int ret;
 	struct sst_platform_data *pdata = pdev->dev.platform_data;
+	struct file *file;
 
 	pr_debug("sst_platform_probe called\n");
+	file = filp_open("/etc/firmware/dfw_sst.bin", O_RDONLY, 0);
+
+	if (IS_ERR(file)) {
+		pr_info("sst_platform_probe is deferred\n");
+		return -EPROBE_DEFER;
+	}
+
 	sst = devm_kzalloc(&pdev->dev, sizeof(*sst), GFP_KERNEL);
 	if (sst == NULL) {
 		pr_err("kzalloc failed\n");
-- 
1.7.9.5

